frontend/src/pages/CompareV2.tsx

```diff
@@ -354,35 +388,97 @@ function CompareV2(props: CompareV2Props) {
         ),
       );
 
-      setConfusionMatrixRunArtifacts(
-        filterRunArtifactsByType(runArtifacts, artifactTypes, MetricsType.CONFUSION_MATRIX)
-          .runArtifacts,
-      );
-      setHtmlRunArtifacts(
-        filterRunArtifactsByType(runArtifacts, artifactTypes, MetricsType.HTML).runArtifacts,
-      );
-      setMarkdownRunArtifacts(
-        filterRunArtifactsByType(runArtifacts, artifactTypes, MetricsType.MARKDOWN).runArtifacts,
-      );
+      // Filter and set the two-panel layout run artifacts.
+      const confusionMatrixRunArtifacts: RunArtifact[] = filterRunArtifactsByType(
+        runArtifacts,
+        artifactTypes,
+        MetricsType.CONFUSION_MATRIX,
+      ).runArtifacts;
+      const htmlRunArtifacts: RunArtifact[] = filterRunArtifactsByType(
+        runArtifacts,
+        artifactTypes,
+        MetricsType.HTML,
+      ).runArtifacts;
+      const markdownRunArtifacts: RunArtifact[] = filterRunArtifactsByType(
+        runArtifacts,
+        artifactTypes,
+        MetricsType.MARKDOWN,
+      ).runArtifacts;
+      setConfusionMatrixRunArtifacts(confusionMatrixRunArtifacts);
+      setHtmlRunArtifacts(htmlRunArtifacts);
+      setMarkdownRunArtifacts(markdownRunArtifacts);
+
+      // Iterate through selected runs, remove current selection if not present among runs.
+      setSelectedArtifactsMap({
+        [MetricsType.CONFUSION_MATRIX]: getVerifiedTwoPanelSelection(
+          confusionMatrixRunArtifacts,
+          selectedArtifactsMap[MetricsType.CONFUSION_MATRIX],
+        ),
+        [MetricsType.HTML]: getVerifiedTwoPanelSelection(
+          htmlRunArtifacts,
+          selectedArtifactsMap[MetricsType.HTML],
+        ),
+        [MetricsType.MARKDOWN]: getVerifiedTwoPanelSelection(
+          markdownRunArtifacts,
+          selectedArtifactsMap[MetricsType.MARKDOWN],
+        ),
+      });
 
+      // Set ROC Curve run artifacts and get all valid ROC curve data for plot visualization.
       const rocCurveRunArtifacts: RunArtifact[] = filterRunArtifactsByType(
         runArtifacts,
         artifactTypes,
         MetricsType.ROC_CURVE,
       ).runArtifacts;
-
-      const { validLinkedArtifacts, fullArtifactPathMap } = getValidRocCurveArtifactData(
-        rocCurveRunArtifacts,
-      );
-
-      setFullArtifactPathMap(fullArtifactPathMap);
-      setRocCurveLinkedArtifacts(validLinkedArtifacts);
-      setSelectedRocCurveIds(
-        validLinkedArtifacts.map(linkedArtifact => getRocCurveId(linkedArtifact)).slice(0, 3),
-      );
+      updateRocCurveDisplay(rocCurveRunArtifacts);
       setIsLoadingArtifacts(false);
+      setIsInitialArtifactsLoad(false);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
The `eslint` warning is disabled as the two-panel layout and ROC Curve tabs are setting variables based on those same existing variables (notably, modifying the `lineColorsStack` and `selectedIdColorMap` in the `updateRocCurveDisplay` method).